libp2p/Host.cpp

```diff
@@ -750,15 +736,9 @@ void Host::startedWorking()
     // initialization (e.g. start capability threads, start TCP listener, and kick off timers)
     asserts(!m_timer);
 
-    {
-        // prevent m_run from being set to true at same time as set to false by stop()
-        // don't release mutex until m_timer is set so in case stop() is called at same
-        // time, stop will wait on m_timer and graceful network shutdown.
-        Guard l(x_runTimer);
-        // create deadline timer
-        m_timer.reset(new io::deadline_timer(m_ioService));
-        m_run = true;
-    }
+    // create deadline timer
+    m_timer.reset(new io::deadline_timer(m_ioService));
+    m_run = true;
```

Comment:
I would move this to right before run `run(boost::system::error_code());` because it's closely related

Reply:
That makes sense to me too...also I think it makes sense since that's after the network is considered to be fully up i.e. the node table has been started, the tcp and udp listeners are up, capability threads are running, etc.

Reply:
Done, but in order for this to work I had to remove the dependency on m_run in `Host::runAcceptor`.